tests/integration/gha-run.sh

```diff
@@ -0,0 +1,153 @@
+}
+
+function cleanup() {
+    platform="${1}"
+
+    if [ "${platform}" = "aks" ]; then
+        az aks delete \
+            -g "kataCI" \
+            -n "${GH_PR_NUMBER}-${GH_PR_SHA}-${KATA_HYPERVISOR}-amd64" \
+            --yes \
+            --no-wait
+        return 0
+    fi
+
+    if [ "${platform}" = "tdx" ]; then
+        deploy_spec="-k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/k3s""
+        cleanup_spec="-k "${tools_dir}/packaging/kata-deploy/kata-cleanup/overlays/k3s""
+    else
+        deploy_spec="-f "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml""
+        cleanup_spec="-f "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml""
+    fi
+
+    kubectl delete ${deploy_spec}
+    kubectl -n kube-system wait --timeout=10m --for=delete -l name=kata-deploy pod
+
+    sed -i -e "s|quay.io/kata-containers/kata-deploy:latest|${DOCKER_REGISTRY}/${DOCKER_IMAGE}:${DOCKER_TAG}|g" "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml"
+    cat "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml"
+    cat "${tools_dir}/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml" | grep "${DOCKER_REGISTRY}/${DOCKER_IMAGE}:${DOCKER_TAG}" || die "Failed to setup the tests image"
+    kubectl apply ${cleanup_spec}
+    sleep 180s
+
+    kubectl delete ${cleanup_spec}
+    kubectl delete -f "${tools_dir}/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml"
+    kubectl delete -f "${tools_dir}/packaging/kata-deploy/runtimeclasses/kata-runtimeClasses.yaml"
+}
+
+function main() {
+    action="${1:-}"
+
+    case "${action}" in
+    install-azure-cli)
+        curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+        # The aks-preview extension is required while the Mariner Kata host is in preview.
+        az extension add --name aks-preview
+        ;;
+    login-azure)
+        az login \
+            --service-principal \
+            -u "${AZ_APPID}" \
+            -p "${AZ_PASSWORD}" \
+            --tenant "${AZ_TENANT_ID}"
+        ;;
+    create-cluster)
+        az aks create \
+            -g "kataCI" \
+            -n "$(_print_cluster_name)" \
+            -s "Standard_D4s_v5" \
+            --node-count 1 \
+            --generate-ssh-keys \
+            $([ "${KATA_HOST_OS}" = "cbl-mariner" ] && echo "--os-sku mariner --workload-runtime KataMshvVmIsolation")
+        ;;
+    install-bats)
+        sudo apt-get update
+        sudo apt-get -y install bats
+        ;;
+    install-kubectl)
+        sudo az aks install-cli
+        ;;
+    get-cluster-credentials)
+        az aks get-credentials \
+            -g "kataCI" \
+            -n "$(_print_cluster_name)"
+        ;;
+    run-tests-aks)
+        run_tests "aks"
+        ;;
+    run-tests-sev)
+        run_tests "sev"
+        ;;
+    run-tests-snp)
+        run_tests "snp"
+        ;;
+    run-tests-tdx)
+        run_tests "tdx"
+        ;;
+    cleanup-aks)
+        cleanup "aks"
```

Comment:
make this its own function as it's different than the others

Reply:
Done